src/Nowruz/modules/dispute/components/contributorDashboard/useContributorDashboard.tsx

```diff
@@ -0,0 +1,139 @@
+import { Cell, ColumnDef, getCoreRowModel, useReactTable } from '@tanstack/react-table';
+import { useEffect, useMemo, useState } from 'react';
+import variables from 'src/components/_exports.module.scss';
+import { Dispute } from 'src/core/api';
+import { disputes, LeaveContribution } from 'src/core/api/disputes/disputes.api';
+import { toRelativeTime } from 'src/core/relative-time';
+import { Chip } from 'src/Nowruz/modules/general/components/Chip';
+import { Dot } from 'src/Nowruz/modules/general/components/dot';
+
+export const useContributorDashboard = (setJoined: (val: boolean) => void) => {
+  const [stopNotif, setStopNotif] = useState(false);
+  const [list, setList] = useState<Dispute[]>([]);
+  const [openModal, setOpenModal] = useState(false);
+
+  useEffect(() => {
+    const getDisputes = async () => {
+      const res = await disputes({ page: 1, limit: 5, 'filter.direction': 'juror' });
```

Comment:
why you didn't put this request in preload on router?
P.S. Fetching data in `useEffect` is not good for tracking and debugging, especially it's in the component level not the page level, and for maintaining we should search nested components until we reach this `useEffect`

Reply:
There is a single route which conditionally renders two different components. if user is joined, 'contributor dashboard' is rendered, otherwise 'join contributor' component is rendered. Data needed for each route is different, so I fetched them in useEffect of related component